src/api.ts

```diff
@@ -1,206 +1,276 @@
-import { Scope, SourceFile, WriterFunctionOrValue, Writers } from 'ts-morph'
-import { OpenAPIV3 } from 'openapi-types'
-import { generateTypeForSchema, writeWriterOrString } from './types'
+import { MethodDeclaration, Scope, SourceFile, WriterFunctionOrValue, Writers } from 'ts-morph'
+import { analyzePathOperations } from './project-analyzer'
+import { writeWriterOrString } from './writer-utils'
+import CodeFormatting from './code-formatting'
+import CodeGen from './code-gen'
+import type { OpenAPIV3 } from 'openapi-types'
+import type { AnalyzedPathItemObject, AnalyzedOperationObject } from './project-analyzer'
 
-const METHODS_WITH_DATA = ['post', 'patch', 'put']
+export interface GenerateApiContext {
+  file: SourceFile
+  spec: OpenAPIV3.Document
+  options: GenerateApiOptions
+}
+
+export interface GenerateApiOptions {
+  removeTagFromOperationId: boolean
+  addReadonlyWriteonlyModifiers: boolean
+}
+
+function getAllReferencedTypes (analyzedPaths: AnalyzedPathItemObject[]) {
```

Comment:
It seems that you miss some types (i.e. the `Company` type in example), you can see errors with the example if you run `ts-node -T cmd/oatyp.ts -s example/openapi.yaml -d example -r true` and check `example/api.ts` file